django/core/cache/backends/base.py

```diff
@@ -266,14 +266,10 @@ def decr_version(self, key, delta=1, version=None):
         return self.incr_version(key, -delta, version)
 
     def touch(self, key, timeout=DEFAULT_TIMEOUT, version=None):
-        """Touch the supplied key and set its expiry time to timeout. Returns True
+        """Touch the supplied key and set its expiry time to timeout. Return True
         if the key was successfully touched, False if the key did not exist.
         """
-        if self.has_key(key, version):
-            self.set(key, self.get(key, version=version), timeout=timeout, version=version)
-            return True
-        else:
-            return False
+        raise NotImplementedError('subclasses of BaseCache must provide a touch() method')
```

Comment:
I don't know if this is a "must", it might not be the case that all cache backends out there can sensibly support it